lib/descriptor-builder.js

```diff
@@ -172,15 +181,18 @@ DescriptorBuilder.prototype.setIdProperty = function(p, validate) {
   this.idProperty = p;
 };
 
-DescriptorBuilder.prototype.assertNotDefined = function(p, name) {
-  var propertyName = p.name,
+DescriptorBuilder.prototype.assertNotDefined = function(p) {
+  var ns = p.ns,
+      propertyName = p.name,
       definedProperty = this.propertiesByName[propertyName];
 
-  if (definedProperty) {
+  if (this.propertiesByName[ns.name] && this.propertiesByName[ns.localName]) {
     throw new Error(
       'property <' + propertyName + '> already defined; ' +
-      'override of <' + definedProperty.definedBy.ns.name + '#' + definedProperty.ns.name + '> by ' +
-      '<' + p.definedBy.ns.name + '#' + p.ns.name + '> not allowed without redefines');
+        'override of <' + definedProperty.definedBy.ns.name + '#' +
```

Comment:
Am I blind or is this exactly equivalent to the original version, just with a different formatting?

Reply:
Sorry, to make the review harder than necessary with this reformatting. In the course of modifying the function, I used code, dismissed it later and forgot to also revert the formatting.

Your remark brought the function back to my mind. After taking another look, I [refactored the variables and changed the error message](https://github.com/bpmn-io/moddle/pull/39/commits/7b0a6f5098b3a1a484515a2cae8f68c2358d8b3c). I hope this makes the namespace aspect more explicit.